server/game/stateWatchers/AttacksThisPhaseWatcher.ts

```diff
@@ -56,9 +56,11 @@ export class AttacksThisPhaseWatcher extends StateWatcher<IAttacksThisPhase> {
         controller: Player;
         filter?: (event: AttackEntry) => boolean;
     }) {
-        return this.getAttackers(
-            (entry) => entry.attackingPlayer === controller && (filter(entry) ?? true)
-        ).length > 0;
+        return this.getAttackers((entry) => {
+            const additionalFilter = (filter !== undefined) ? filter(entry) : true;
```

Comment:
When doing checks like this we need to use `==` and `!=` as those will match either null or undefined. It's basically the only time it's right to use those instead of `===` and `!==`

Reply:
gotcha